### ISSUE TITLE
Lutris 0.4.5 doesn’t launch (AssertionError)

### ISSUE THREAD

**Issue by lmintmate** [lutris/lutris#525]
I'm using Linux Mint 18.1 (my de is Mate if that's any help) and I received the update for Lutris 0.4.5. I tried to launch Lutris after that and nothing seemed to happen. I then tried to launch it from the terminal with the command "lutris" and saw the following:

```python
Traceback (most recent call last):
File "/usr/lib/python3/dist-packages/lutris/gui/application.py", line 152, in do_activate
self.window = LutrisWindow(application=self)
File "/usr/lib/python3/dist-packages/lutris/gui/lutriswindow.py", line 84, in init
desktopapps.sync_with_lutris()
File "/usr/lib/python3/dist-packages/lutris/util/desktopapps.py", line 87, in sync_with_lutris
mark_as_installed(appid, 'linux', game_info)
File "/usr/lib/python3/dist-packages/lutris/util/desktopapps.py", line 32, in mark_as_installed
assert game_info[key]
AssertionError
```

I do hope you fix it, as I run quite a few dosbox/scummvm games, and a Wine game with it, so I kind of need it for gaming. Thanks for the understanding.
P.S. I had also posted it on the forum, but overcame my reluctance to not creating many accounts online, as it's kind of important that this gets resolved.

**Comment by daniel-j** [comment]
I'll have a look at this as soon as I can, thanks for reporting it! :)

**Comment by daniel-j** [comment]
Try this and see if it works: https://github.com/daniel-j/lutris/tree/fix-desktopapps

Run Lutris like this from the repo root folder: `bin/lutris -d`

Please report back what desktop games it logs out and what it fails to load.

**Comment by lmintmate** [comment]
Thanks for taking the time to look at my problem. It's the first time I follow a fix like this, so I hope I got it right:

I pressed the "clone or download" button and downloaded the lutris-fix-desktopapps.zip I then unzipped the file and went into the bin directory where I saw these 2 files, lutris and padsp32. I wasn't exactly certain how to run the bin/lutris -d command as you told me (the result I had had something to do with the bin being a directory), so I just double clicked the Lutris file and had it run through the terminal. It showed up various stuff being loaded and then Lutris thankfully ran. Because I wanted to be sure I saw correctly I then open the bin directory in the terminal and wrote ./lutris -d, upon which I saw these:

```
DEBUG    2017-03-09 11:46:13,183 [desktopapps]:Found desktop game "Bio Menace" (gog_com-Bio_Menace_1.desktop)
DEBUG    2017-03-09 11:46:13,184 [desktopapps]:Found desktop game "Retro City Rampage DX" (gog_com-Retro_City_Rampage_DX_1.desktop)
DEBUG    2017-03-09 11:46:13,184 [desktopapps]:Found desktop game "Gemini Rue" (gog_com-Gemini_Rue_1.desktop)
DEBUG    2017-03-09 11:46:13,184 [desktopapps]:Found desktop game "PyChess" (pychess.desktop)
DEBUG    2017-03-09 11:46:13,184 [desktopapps]:Found desktop game "Gweled" (gweled.desktop)
DEBUG    2017-03-09 11:46:13,184 [desktopapps]:Found desktop game "Battle for Wesnoth Map Editor (1.12)" (wesnoth-1.12_editor.desktop)
DEBUG    2017-03-09 11:46:13,184 [desktopapps]:Found desktop game "Kega Fusion" (kega-fusion.desktop)
ERROR    2017-03-09 11:46:13,184 [desktopapps]:Failed to load desktop game "Νάρκες" (gnome-mines.desktop)
DEBUG    2017-03-09 11:46:13,184 [desktopapps]:Found desktop game "Chocolate Doom" (chocolate-doom.desktop)
DEBUG    2017-03-09 11:46:13,185 [desktopapps]:Found desktop game "Sam & Max Hit the Road" (gog_com-Sam_Max_Hit_the_Road_1.desktop)
DEBUG    2017-03-09 11:46:13,185 [desktopapps]:Found desktop game "Hedgewars" (hedgewars.desktop)
DEBUG    2017-03-09 11:46:13,185 [desktopapps]:Found desktop game "Πασιέντζα AisleRiot" (sol.desktop)
DEBUG    2017-03-09 11:46:13,185 [desktopapps]:Found desktop game "Fish Fillets" (fillets-ng.desktop)
DEBUG    2017-03-09 11:46:13,185 [desktopapps]:Found desktop game "UFO: Alien Invasion" (ufoai.desktop)
DEBUG    2017-03-09 11:46:13,185 [desktopapps]:Found desktop game "Defender's Quest" (gog_com-Defenders_Quest_1.desktop)
DEBUG    2017-03-09 11:46:13,185 [desktopapps]:Found desktop game "Dune Legacy" (dunelegacy.desktop)
DEBUG    2017-03-09 11:46:13,185 [desktopapps]:Found desktop game "Frogatto" (frogatto.desktop)
DEBUG    2017-03-09 11:46:13,186 [desktopapps]:Found desktop game "Hacknet" (gog_com-Hacknet_1.desktop)
DEBUG    2017-03-09 11:46:13,186 [desktopapps]:Found desktop game "Gargoyle" (gargoyle.desktop)
DEBUG    2017-03-09 11:46:13,186 [desktopapps]:Found desktop game "FreeCol" (freecol.desktop)
DEBUG    2017-03-09 11:46:13,186 [desktopapps]:Found desktop game "SuperTux 2" (supertux2.desktop)
DEBUG    2017-03-09 11:46:13,186 [desktopapps]:Found desktop game "Freeciv server" (freeciv-server.desktop)
DEBUG    2017-03-09 11:46:13,186 [desktopapps]:Found desktop game "Magnetic Scrolls interpreter" (gtkmagnetic.desktop)
DEBUG    2017-03-09 11:46:13,186 [desktopapps]:Found desktop game "Freedroid RPG" (freedroidrpg.desktop)
DEBUG    2017-03-09 11:46:13,186 [desktopapps]:Found desktop game "CLARC" (gog_com-CLARC_1.desktop)
DEBUG    2017-03-09 11:46:13,187 [desktopapps]:Found desktop game "Freeciv (gtk2)" (freeciv-gtk2.desktop)
DEBUG    2017-03-09 11:46:13,187 [desktopapps]:Found desktop game "Angband (SDL)" (angband-sdl.desktop)
DEBUG    2017-03-09 11:46:13,187 [desktopapps]:Found desktop game "Angband (X11)" (angband-x11.desktop)
DEBUG    2017-03-09 11:46:13,187 [desktopapps]:Found desktop game "OpenXcom" (openxcom.desktop)
DEBUG    2017-03-09 11:46:13,187 [desktopapps]:Found desktop game "Battle for Wesnoth (1.12)" (wesnoth-1.12.desktop)
DEBUG    2017-03-09 11:46:13,187 [desktopapps]:Found desktop game "HuniePop" (gog_com-HuniePop_1.desktop)
DEBUG    2017-03-09 11:46:13,187 [desktopapps]:Found desktop game "Freeciv modpack installer (gtk3)" (freeciv-mp-gtk3.desktop)
DEBUG    2017-03-09 11:46:13,187 [desktopapps]:Found desktop game "SuperTuxKart" (supertuxkart.desktop)
DEBUG    2017-03-09 11:46:13,188 [desktopapps]:Found desktop game "Frozen-Bubble" (frozen-bubble.desktop)
DEBUG    2017-03-09 11:46:13,308 [system]:Executing which steam
DEBUG    2017-03-09 11:46:14,090 [runtime]:Runtime steam is up to date
DEBUG    2017-03-09 11:46:14,091 [runtime]:Runtime lib32 is up to date
DEBUG    2017-03-09 11:46:14,091 [runtime]:Runtime p7zip is up to date
DEBUG    2017-03-09 11:46:14,092 [runtime]:Runtime lib64 is up to date
DEBUG    2017-03-09 11:46:14,098 [api]:Fetching game library
```

Upon repeating the execution of Lutris by terminal I narrowed down the culprit:

`ERROR    2017-03-09 11:48:22,158 [desktopapps]:Failed to load desktop game "Νάρκες" (gnome-mines.desktop)`

It thus looks like the auto-import feature fails to load gnome-mines, and this might have to do with the fact the name this package has in the start menu is in Greek (I'm Greek and my Linux Mint install is in Greek too, so this might have been why you couldn't reproduce the issue), but that's just a wild guess of mine, especially as it didn't fail with Aisleriot whose name was also partially Greek.

**Comment by daniel-j** [comment]
Ok, I think that's it then. I don't log it out right now but internally Lutris creates a "slug" of the game name. For example "Dune Legacy" would become "dune-legacy". Seems like this doesn't work with greek, as the slug can only have contain ascii characters! I'll add a fix so it can make the slug of the filename (gnome-mines) instead if the name fails.

**Comment by daniel-j** [comment]
@lmintmate I have updated the code, download zip again! Νάρκες should appear in Lutris! :)

**Comment by lmintmate** [comment]
I just executed the lutris executable from the new zip file and now it doesn't show any error messages when I launch from the terminal - it only showed `INFO     2017-03-09 12:57:52,320 [desktopapps]:Setting Νάρκες as installed` and then launched. When I scrolled through my games I can now see Νάρκες as well, and I can also launch it.
Thanks for your help! Hopefully your fix can be merged with the main branch of Lutris so that I (and others that might have gotten this problem) will be able to launch Lutris from the regular start menu shortcut as well, and not only from the executable included with this particular zip file.
